packages/cyberstorm/src/components/Forms/CreateTeamForm/CreateTeamForm.tsx

```diff
@@ -0,0 +1,118 @@
+"use client";
+import styles from "./CreateTeamForm.module.css";
+import * as Button from "../../Button";
+import { TextInput } from "../../TextInput/TextInput";
+import { useForm, FieldErrors, useController } from "react-hook-form";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faArrowsRotate } from "@fortawesome/free-solid-svg-icons";
+
+import { useContext } from "react";
+import { ToastContext } from "../../Toast/ToastContext";
+import {
+  faCircleCheck,
+  faCircleExclamation,
+  faOctagonExclamation,
+} from "@fortawesome/pro-solid-svg-icons";
+
+/**
+ * Form for creating a team
+ */
+export function CreateTeamForm() {
+  const useToast = () => useContext(ToastContext);
+  const toast = useToast();
+
+  async function onValid(data: { teamName: string }) {
+    // TODO: Add sending to TS API endpoint
+    toast?.addToast(
+      "info",
+      <FontAwesomeIcon icon={faCircleExclamation} />,
+      "Creating team"
+    );
+    await new Promise((resolve) => {
+      setTimeout(resolve, 1000);
+    });
+    // TODO: Add a toast on response return based on the response
+    toast?.addToast(
+      "success",
+      <FontAwesomeIcon icon={faCircleCheck} />,
+      "Team created"
+    );
+    console.log(JSON.stringify(data));
+  }
+
+  async function onInvalid(
+    errors: FieldErrors<{
+      teamName: string;
+    }>
+  ) {
+    if (errors.teamName) {
+      toast?.addToast(
+        "danger",
+        <FontAwesomeIcon icon={faOctagonExclamation} />,
+        errors.teamName.message,
+        30000
+      );
+    } else {
+      toast?.addToast(
+        "danger",
+        <FontAwesomeIcon icon={faOctagonExclamation} />,
+        "Unknown error",
+        30000
+      );
+    }
+  }
+
+  const {
+    control,
+    handleSubmit,
+    formState: { isSubmitting },
+  } = useForm<{ teamName: string }>({
+    mode: "onSubmit",
+  });
+
+  const teamName = useController({
+    control: control,
+    name: "teamName",
+    rules: {
+      required: "Team name is required",
+    },
+  });
+
+  return (
+    <form
```

Comment:
There's no commit message and the linked tasks don't really tell me anything so it's a bit hard to know which level of feedback would be needed here. So in short, from the top of my mind IDK what would be the best way to handle this, but this approach seems very verbose and manual. Maybe it's what we were aiming for, for now? I'm just a bit worried that since this is the first form, it will get copy-pasted around, creating code debt.

Reply:
I think the new version in fix commit, is a lot better 🤷